lms/lms/lms/private.rs

```diff
@@ -0,0 +1,354 @@
+use crate::constants::{D_INTR, D_LEAF, ID_LEN};
+use crate::error::LmsDeserializeError;
+use crate::lms::{LmsMode, PublicKey, Signature};
+use crate::ots::PrivateKey as OtsPrivateKey;
+use crate::types::{Identifier, Typecode};
+
+use anyhow::anyhow;
+use digest::{Digest, Output, OutputSizeUser};
+use generic_array::{ArrayLength, GenericArray};
+use rand::{CryptoRng, Rng};
+use signature::{Error, RandomizedSignerMut};
+
+use std::cmp::Ordering;
+use std::ops::Add;
+use typenum::{Sum, U28};
+
+/// Opaque struct representing a LMS public key
+///
+/// Note: there is no requirement to map specific LMS algorithms to specific
+/// LM-OTS algorithms so it must be parametrized. With the algorithms provided
+/// by this crate, this is done via
+/// [LmsSha256M32H10](crate::lms::LmsSha256M32H10)<[LmsOtsSha256N32W4](crate::ots::LmsOtsSha256N32W4)>.
+pub struct PrivateKey<Mode: LmsMode> {
```

Comment:
As a recommendation for cross-crate consistency, we generally use the names `SigningKey` and `VerifyingKey`, which are named after their roles / what they do, a convention also adopted in `ed25519-dalek`

Reply:
I have updated the names to `SigningKey` and `VerifyingKey`